Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -327,15 +327,29 @@ ROMol *replaceCore(const ROMol &mol, const ROMol &coreQuery,
     return 0;
   }
 
+  return replaceCore(mol, coreQuery, matchV,
+                     replaceDummies, labelByIndex,
+                     requireDummyMatch);
+}
+
+ROMol *replaceCore(const ROMol &mol,
+                   const ROMol &core,
+                   const MatchVectType &matchV,
+                   bool replaceDummies,
+                   bool labelByIndex,
+                   bool requireDummyMatch) {
   unsigned int origNumAtoms = mol.getNumAtoms();
   std::vector<int> matchingIndices(origNumAtoms, -1);
   std::vector<int> allIndices(origNumAtoms, -1);
-  //  std::cerr << "replaceDummies " << (int) replaceDummies << std::endl;
   for (MatchVectType::const_iterator mvit = matchV.begin();
        mvit != matchV.end(); mvit++) {
+    PRECONDITION( static_cast<unsigned int>(mvit->first) < core.getNumAtoms(),
```

Comment:
The tests are good, but I think they should throw ValueErrors instead of using the Invariant macros.